src/test/ui/consts/const-eval/mod-static-with-const-fn.rs

```diff
@@ -27,9 +27,6 @@ fn foo() {}
 
 static BAR: () = unsafe {
     *FOO.0.get() = 5;
-    //~^ ERROR statements in statics are unstable (see issue #48821)
-    // This error is caused by a separate bug that the feature gate error is reported
-    // even though the feature gate "const_let" is active.
```

Comment:
The reason we do not get an error at all here any more is that the static detection misses it, and the dynamic one doesn't get there because we don't run this constant, because of the call to `foo()` later -- right?

If so, please add a comment.

Reply:
Btw, I find it quite curious that here we are not executing the MIR after it was deemed illegal, unlike the cases I can into with https://github.com/rust-lang/rust/pull/56007.